Uyanda.Coffee.Persistence/Accessors/BeverageAccessor.cs

```diff
@@ -67,6 +67,56 @@ public async Task<InvoiceModel> PurchaseAsync(IEnumerable<LineItemModel> lineIte
             return ToModel(invoice);
         }
 
+        public async Task<BeverageSizeCostModel[]> AddBeverageSizeCostAsync(BeverageSizeCostModel price)
+        {
+          
+            var beverageDictionary = await localDbContext.Beverages.AsNoTracking()
```

Comment:
There is a difference between _server side_ and _client side_ processing. If you can, prefer _server side_.  In this intance your PC is a client and the DB server (for now it happens to be your PC also) is the server. So you go fetch everything from the server (it will also use a lot more bandwidth and have latency) and then interogate the result locally (the _if_ on line 76). Rather have a _.Where_ clause in this LINQ query. That way, the SQL server will do the interogation and return only the result  you want (saving bandwidth and have less latancy).